x-pack/plugins/task_manager/server/task_pool.ts

```diff
@@ -227,14 +228,17 @@ export class TaskPool {
     }
   }
 
-  private async cancelTask(task: TaskRunner) {
-    try {
-      this.logger.debug(`Cancelling task ${task.toString()}.`);
-      this.tasksInPool.delete(task.taskExecutionId);
-      await task.cancel();
-    } catch (err) {
-      this.logger.error(`Failed to cancel task ${task.toString()}: ${err}`);
-    }
+  private cancelTask(task: TaskRunner) {
+    // internally async (without rejections), but public-facing is synchronous
+    void (async () => {
```

Comment:
This is the perfect example of what I was talking in https://github.com/elastic/kibana/pull/181456#discussion_r1584314225: it's a very controlled scenario where we know it'll never reject, so adding a `.catch` to it is adding a branch to the code that will never be executed. Since we don't want to await the promise, we can use `void` to tell the linter "it's fine to leave this unhandled".

Reply:
Ya, I think using `.catch()` over `void` is going to be my default for these ...

Reply:
Deal! Changed in [`1e7f5de` (#181456)](https://github.com/elastic/kibana/pull/181456/commits/1e7f5de3f93281f33455b8fa201600efb634461d)